src/compiler/parsing/watf/number-literals.js

```diff
@@ -51,6 +52,14 @@ export function parse32I(sourceString: string): number {
   return value;
 }
 
+export function parseU32(sourceString: string): number {
+  const value = parse32I(sourceString);
+  if (value < 0) {
```

Comment:
I would use `Math.sign(value) !== 1` instead. It's closer to an actual sign check I think and handles cases like NaN.